scaler/handlers.go

```diff
@@ -38,18 +38,39 @@ func (e *impl) IsActive(
 	ctx context.Context,
 	scaledObject *externalscaler.ScaledObjectRef,
 ) (*externalscaler.IsActiveResponse, error) {
+	lggr := e.lggr.WithName("IsActive")
+	host, ok := scaledObject.ScalerMetadata["host"]
+	if !ok {
+		err := fmt.Errorf("no 'host' field found in ScaledObject metadata")
+		lggr.Error(err, "ScaledObjectRef", scaledObject)
+		return nil, err
+	}
+	if host == "interceptor" {
+		return &externalscaler.IsActiveResponse{
+			Result: true,
+		}, nil
```

Comment:
To avoid this repeated return, what about creating a separated `func` to execute the logic? Like:

```go
func isActiveCheck (allCounts map[string]int, host string) bool {
	if host === "interceptor" {
      return true
    }
    hostCount, ok := allCounts[host]

    return ok && hostCount > 0 ? true : false
}
```

Reply:
I think it would be a bit over-engineering at the moment. We can do it in the next iteration when we plan to do the push from the interceptor.

Reply:
Also we wouldn't be calling this function in a lot of places right now right ? So we don't have to abstract this logic right ?

Reply:
I think that, in the case of not setting the `hostCount` to 0, we wouldn't have a big problem. Because there were to be only two return statements, and they're both different from each other, since the first one is an early return to avoid scaling the interceptor down. The final idea I could think of was:

```go
func (e *impl) IsActive(
	ctx context.Context,
	scaledObject *externalscaler.ScaledObjectRef,
) (*externalscaler.IsActiveResponse, error) {
	lggr := e.lggr.WithName("IsActive")
	host, ok := scaledObject.ScalerMetadata["host"]
	if !ok {
    err := fmt.Errorf("no 'host' field found in ScaledObject metadata")
		lggr.Error(err, "returning immediately from IsActive RPC call", "ScaledObject", *scaledObject)
		return nil, err
	}

	allCounts := e.pinger.counts()
	active, err := isActive(host, allCounts)
	if err != nil {
		err := fmt.Errorf("host '%s' not found in counts", host)
		lggr.Error(err, "Given host was not found in queue count map", "host", host, "allCounts", allCounts)
		return nil, err
	}

	return &externalscaler.IsActiveResponse{
		Result: active,
	}, nil
}

func isActive (host string, allCounts map[string]int) (bool, error) {
	if host == "interceptor" {
		return true
	}

	hostCount, ok := allCounts[host]
	if !ok {
		return false, fmt.Errorf("host '%s' not found in counts", host)
	}

	return hostCount > 0, nil
}
```

But this doesn't add much, so it's better to stick with this implementation.